app/services/comparison_service.rb

```diff
@@ -54,9 +54,11 @@ def refresh_comparisons
       new_snapshot = store_new_snapshot!(project, result, refreshed_at)
       clean_up_old_snapshots
     end
-    project.stages.select { |s| warrants_deploy?(s) }.each do |stage|
-      stage.deploy_strategies.each do |strategy|
-        DeployService.start(strategy) if strategy.automatic?
+    unless project.deploy_blocks.unresolved.any?
```

Comment:
slight preference for `if project.deploy_blocks.unresolved.empty?` to keep things more positive, bit easier to understand IMO.

Reply:
I actually had this as `if ... unresolved.none?` initially but the double negative felt bad. I think I'll go with your suggestion.